vsphere/vsphere.go

```diff
@@ -387,13 +393,49 @@ func (ops *vsphereOps) FreeDevices(blockDeviceMappings []interface{}, rootDevice
 	}
 }
 
-func (ops *vsphereOps) Inspect(diskPaths []*string) ([]interface{}, error) {
-	// TODO find a way to map diskPaths to unattached/attached virtual disks and query info
-	// currently returning the disks directly
+func (ops *vsphereOps) Inspect(vmdksWithDS []*string) ([]interface{}, error) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
 
-	return nil, &cloudops.ErrNotSupported{
-		Operation: "Inspect",
+	vmObj, err := ops.renewVM(ctx, ops.vm)
+	if err != nil {
+		return nil, err
+	}
+	f := find.NewFinder(vmObj.Client(), true)
+	f.SetDatacenter(vmObj.Datacenter.Datacenter)
+
+	dsMap := make(map[string]*object.Datastore)
+	disks := []interface{}{}
+	for _, vmdkPathWithDS := range vmdksWithDS {
+		matches := vmdkMatcherRegex.FindStringSubmatch(*vmdkPathWithDS)
+		if len(matches) < 2 {
+			return nil, fmt.Errorf("failed to inspect drive: "+
+				"failed to parse datastore from vmdk path: %s", *vmdkPathWithDS)
+		}
+		dsName := matches[1]
+		vmdkPath := strings.TrimSpace(matches[2])
+
+		ds, exists := dsMap[dsName]
+		var err error
+		if !exists {
+			ds, err = f.Datastore(context.TODO(), dsName)
+			if err != nil {
+				return nil, fmt.Errorf("failed to inspect drive: failed to "+
+					"get datastore %s for vmdk %s: %s", dsName, vmdkPath, err)
+			}
+			dsMap[dsName] = ds
+		}
+		m := object.NewVirtualDiskManager(ds.Client())
+		diskInfos, err := m.QueryVirtualDiskInfo(ctx, *vmdkPathWithDS, vmObj.Datacenter.Datacenter, false)
+		if err != nil {
+			return nil, fmt.Errorf("failed to inspect drive %v: %v", *vmdkPathWithDS, err)
+		}
+		if len(diskInfos) != 1 {
+			return nil, fmt.Errorf("failed to inspect drive %v: found more than 1 disks with the same name", *vmdkPathWithDS)
```

Comment:
nit: change to `found more %d disks...`,  and print len(diskInfos)..

error would be misleading if somehow it returns length 0.